build-tests/api-documenter-scenarios/src/runScenarios.ts

```diff
@@ -1,154 +1,85 @@
+            throw e;
+          }
         }
-      },
-      compilerState
-    });
-
-    if (extractorResult.errorCount > 0) {
-      anyErrors = true;
-    }
-
-    // API Documenter will always look for a config file in the same place (it cannot be configured), so this script
-    // manually overwrites the API documenter config for each scenario. This is in contrast to the separate config files
-    // created when invoking API Extractor above.
-    const apiDocumenterOverridesPath = path.resolve(
-      `./src/${scenarioFolderName}/config/api-documenter-overrides.json`
-    );
-    const apiDocumenterJsonOverrides = FileSystem.exists(apiDocumenterOverridesPath)
-      ? JsonFile.load(apiDocumenterOverridesPath)
-      : {};
-    const apiDocumenterJson = {
-      $schema: 'https://developer.microsoft.com/json-schemas/api-extractor/v7/api-documenter.schema.json',
-      outputTarget: 'markdown',
-      tableOfContents: {},
-      ...apiDocumenterJsonOverrides
-    };
-
-    JsonFile.save(apiDocumenterJson, apiDocumenterJsonPath, { ensureFolderExists: true });
 
-    // TODO: Ensure that the checked-in files are up-to-date
-    // Run the API Documenter command-line
-    executeCommand(
-      'node node_modules/@microsoft/api-documenter/lib/start ' +
-        `generate --input-folder etc/${scenarioFolderName} --output-folder etc/${scenarioFolderName}/markdown`
-    );
-  }
-
-  // Delete the transient `api-documenter.json` file before completing, as it'll just be whatever the last scenario
-  // was, and shouldn't be committed.
-  FileSystem.deleteFile(apiDocumenterJsonPath);
-
-  if (!anyErrors) {
-    process.exitCode = 0;
+        const apiDocumenterJson: {} = {
+          $schema: 'https://developer.microsoft.com/json-schemas/api-extractor/v7/api-documenter.schema.json',
+          outputTarget: 'markdown',
+          tableOfContents: {},
+          ...apiDocumenterJsonOverrides
+        };
+
+        await JsonFile.saveAsync(apiDocumenterJson, apiDocumenterJsonPath, { ensureFolderExists: true });
+
+        // TODO: Ensure that the checked-in files are up-to-date
+        // Run the API Documenter command-line
+        const childProcess: ChildProcess = Executable.spawn(
+          process.argv0,
+          [
+            'node_modules/@microsoft/api-documenter/lib/start',
+            'generate',
+            `--input-folder`,
+            `temp/etc/${scenarioFolderName}`,
+            '--output-folder',
+            `temp/etc/${scenarioFolderName}/markdown`
+          ],
+          {
+            stdio: 'inherit'
+          }
+        );
+
+        await Executable.waitForExitAsync(childProcess, {
+          throwOnNonZeroExitCode: true,
+          throwOnSignal: true
+        });
+      }
+    });
+  } finally {
+    // Delete the transient `api-documenter.json` file before completing, as it'll just be whatever the last scenario
+    // was, and shouldn't be committed.
+    await FileSystem.deleteFileAsync(apiDocumenterJsonPath);
```

Comment:
Wrap this in a try-catch to not swallow if its a not-exists error. Keeps logs clean in failures earlier on in the build process

Reply:
The default is to not throw if it already doesn't exist.